tests/shared/src/test/scala/scala/meta/tests/parsers/ScaladocParserSuite.scala

```diff
@@ -206,6 +206,44 @@ class ScaladocParserSuite extends FunSuite {
     assertEquals(result, expectation)
   }
 
+  test("markdown code blocks") {
+    val codeBlock = // keep all newlines and leading spaces
+      """|  object Foo {
+         |    def bar(): String = "foobar"
+         |  }
+         |""".stripMargin.split("\n")
+    val codeBlockAsComment = codeBlock.mkString("\n *")
+
+    val result =
+      parseString(
+        s"""
+          /**
+            *  ```scala sc:compile
+            *$codeBlockAsComment
+            *  ```
```

Comment:
what happens if closing quotes are on the last line?

Reply:
I've added a test case for it. The last line of the markdown code block should be just 3 backticks. They can't be on the same line as the code.

Reply:
@tdudzik any new scaladoc features we implement need to be clearly documented first. otherwise, people will keep filing issues about something that they expected to work, based on a random interpretation or some thread they read somewhere. for instance, you mentioned `sc:compile` but i have no idea what it does, and don't know where to find out.

so, documentation should be the first step, before this change. @tgodzik thoughts?

the link you shared before, https://github.com/lampepfl/dotty/tree/master/scaladoc, contains no specific information on what is supported (just says "markup"). all the `virtuslab` links it contains produce a "resource not found" error. i'd like to see all elements documented and explained, at least as well as in https://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html which is what was used for the current scaladoc parser implementation. (keep in mind that there's another implementation in scalameta, under contrib, which doesn't fully follow that page and is therefore a bit less structured.)

Reply:
I think we should fix the docs for sure, I will talk to the team responsible. I think scaladoc for Scala 3 follows the same guidelines as the one for Scala 3 plus Markdown code blocks, which are actually documented, so this wouldn't be a surprise to the users.

Reply:
We can wait with with the merge until scaladoc is better documented and concurrently see if it's actually possible to format code blocks in scalafmt using the wiki syntax `{{{`

Reply:
Hi @kitbellew, here is [PR](https://github.com/lampepfl/dotty/pull/12083) adding docs for scaladoc.
To have a quick preview how these docs would look like after merge look at [rendered docs](https://scala3doc.virtuslab.com/pr-update-docs/scala3/docs/usage/scaladoc/index.html) or simple README under `scaladoc/` [directory](https://github.com/lampepfl/dotty/blob/0288354bf504130ca546313d3abc58ac0f759deb/scaladoc/README.md)
I'd love to hear your opinion and what sections do we should cover.

As it comes to `sc:compile` this part of work is not yet documented, as it's experimental feature developed by @pikinier20 and is subject to changes, but it's related to snippet compiler about which you can read more [here](https://contributors.scala-lang.org/t/snippet-validation-in-scaladoc-for-scala-3/4976)

Please let us know what we could improve

Reply:
@barkingbad thank you, i made a small comment on that pr.

@tdudzik found some documentation for commonmark, seems that we need to support `` ``` `` with or without `scala` label. in fact, it's plausible that the code comes from another language as an example of what the code below the scaladoc had been based on.

so, why don't we indeed add a new `MDCodeBlock` class, with optional language, optional parameters a-la `sc:compile`, and some code lines (preserving indentation)?